Neos.ContentGraph.PostgreSQLAdapter/src/Domain/Projection/Feature/NodeReferencing.php

```diff
@@ -50,24 +50,25 @@ function (NodeRecord $node) {
                     }
                 );
 
-                // remove old
-                $this->getDatabaseConnection()->delete($this->tableNamePrefix . '_referencerelation', [
-                    'sourcenodeanchor' => $anchorPoint->value,
-                    'name' => $event->referenceName->value
-                ]);
-
-                // set new
                 $position = 0;
-                foreach ($event->references as $reference) {
-                    $referenceRecord = new ReferenceRelationRecord(
-                        $anchorPoint,
-                        $event->referenceName,
-                        $position,
-                        $reference->properties,
-                        $reference->targetNodeAggregateId
-                    );
-                    $referenceRecord->addToDatabase($this->getDatabaseConnection(), $this->tableNamePrefix);
-                    $position++;
+                foreach ($event->references as $referencesForProperty) {
+                    $this->getDatabaseConnection()->delete($this->tableNamePrefix . '_referencerelation', [
+                        'sourcenodeanchor' => $anchorPoint->value,
+                        'name' => $referencesForProperty->referenceName->value
+                    ]);
+
+                    foreach ($referencesForProperty->references as $reference) {
+                        // set new
+                        $referenceRecord = new ReferenceRelationRecord(
+                            $anchorPoint,
+                            $referencesForProperty->referenceName,
+                            $position,
+                            $reference->properties,
+                            $reference->targetNodeAggregateId
+                        );
+                        $referenceRecord->addToDatabase($this->getDatabaseConnection(), $this->tableNamePrefix);
+                        $position++;
+                    }
```

Comment:
Same here, can't we turn this into two atomic queries?

Reply:
for me the pg adapter is out of scope and there are no tests ... but i fixed it for doctrine